sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -674,16 +695,26 @@ def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None):
                     sampled = ice_lines[np.random.choice(
                         ice_lines.shape[0], n_samples, replace=False
                     ), :]
+                    if centered:
+                        sampled -= sampled[:, 0, None]
                     for j, ins in enumerate(sampled):
                         lines_ravel[i * j + j] = axi.plot(
```

Comment:
Th indexer `[i * j + j]` is faulty. It overwrites some lines, while leaving out others, resulting in None. If you want to have the PDPs  ~~in the first elements of the raveled view, it should be `[n_cols + (j+n_samples * i) ]`~~
in the first element of the second dim of the *un*raveled lines (which is as far as I can guess what you initially wanted to do?), that is f.i. in `self.lines_[0, :, 0]`, then the indexer must be `[i * n_samples + i + j + 1]`.

Reply:
Could you open another issue with a non-regression test

Reply:
Yep, going to do that.